File: voice_talk_ios/Bridge/Components/DeviceAuthComponent.swift
//
//  DeviceAuthComponent.swift
//  voice_talk_ios
//
//  Created for IDFV-based device authentication
//

import HotwireNative
import UIKit

final class DeviceAuthComponent: BridgeComponent {
    override class var name: String { "device-auth" }
    
    override func onReceive(message: Message) {
        switch message.event {
        case "getDeviceId":
            handleGetDeviceId(message: message)
        default:
            break
        }
    }
    
    private func handleGetDeviceId(message: Message) {
        // IDFV 가져오기
        if let idfv = UIDevice.current.identifierForVendor?.uuidString {
            print("✅ IDFV obtained: \(idfv)")
            reply(to: message.event, with: ["deviceId": idfv])
        } else {
            print("❌ IDFV not available")
            reply(to: message.event, with: ["error": "IDFV not available"])
        }
    }
}

File: android/app/src/main/java/com/example/voice_talk_android/bridge/DeviceAuthComponent.kt
package com.example.voice_talk_android.bridge

import android.provider.Settings
import dev.hotwire.core.bridge.BridgeComponent
import dev.hotwire.core.bridge.BridgeDelegate
import dev.hotwire.core.bridge.Message
import dev.hotwire.navigation.destinations.HotwireDestination

class DeviceAuthComponent(
    name: String,
    private val delegate: BridgeDelegate<HotwireDestination>
) : BridgeComponent<HotwireDestination>(name, delegate) {

    override fun onReceive(message: Message) {
        when (message.event) {
            "getDeviceId" -> handleGetDeviceId(message)
        }
    }

    private fun handleGetDeviceId(message: Message) {
        try {
            val context = delegate.destination.fragment.requireContext()
            val androidId = Settings.Secure.getString(
                context.contentResolver,
                Settings.Secure.ANDROID_ID
            )
            android.util.Log.d("DeviceAuthComponent", "✅ Android ID obtained: $androidId")
            replyTo(message.event, mapOf("deviceId" to androidId))
        } catch (e: Exception) {
            android.util.Log.e("DeviceAuthComponent", "❌ Failed to get Android ID: ${e.message}")
            replyTo(message.event, mapOf("error" to e.message))
        }
    }
}

